docs/389ds/design/accesslog-json-design.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Access Log JSON Format Design"
+---
+
+# Access Log JSON Format
+----------------
+
+Overview
+--------
+
+In order to make the access logging more consumable by standard parsing tools there should be an option for writing the log in JSON
+
+JSON Design
+------------
+
+    {
+        // Header
+        version: "",
+        instance: "",
+        // The rest of the log
+        op: (
+            ABANDON, ADD, AUTOBIND, BIND/UNBIND, CLOSE, CMP, CONN, DEL, EXT,
+            MOD, MODRDN, SEARCH, SORT, STAT, RESULT, VLV
```

Comment:
I wonder if we would not want to add a generic CTRL op to log incoming control
  with the OID and the criticality flags (I am more dubious about the logging the control data because it may contains secret - that said we may have a white list of control whose data are safe to log)
  
And maybe another op for  HTTP PROXY operation ?

Reply:
About controls: maybe a field with a sequence of 
    "control"  object containing: criticallity, oid, optionnal string representing the data
   (probably cheaper than adding a new record for each controls)

Reply:
And reuse the same field/object for the response/control in RESULT records
If that is the case we may remove the need fort VLV/SORT tags since the information will be provided by the generic format

Reply:
Sorry I'm not sure I'll following your idea.  Controls could be added to the operations in which they were attached to.  I don't see how it would be its own log "event".  Sorry I think I am just misunderstanding your suggestion.

Reply:
Ok at first I thought you wanted it to be its own event.  Anyway I fully agree about adding controls to the logging :+1: